runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -193,14 +194,17 @@ local function on_line_impl(self, buf, line)
       local start_row, start_col, end_row, end_col = node:range()
       local hl = highlighter_query.hl_cache[capture]
 
-      if hl and end_row >= line then
+      local is_spell = highlighter_query:query().captures[capture] == 'spell'
+
+      if hl and end_row >= line and (not spell or is_spell) then
         a.nvim_buf_set_extmark(buf, ns, start_row, start_col, {
           end_line = end_row,
           end_col = end_col,
           hl_group = hl,
```

Comment:
```suggestion
          hl_group = not is_spell and hl or nil,
```

Reply:
Really? I think that we can combine both.

Reply:
There is no `@spell` highlight.

Reply:
But I guess, having this set to `@spell` isn't harmful since it'll be cleared by default, and allows users to easily see which regions are being spell checked by setting the group.

Reply:
Oh I see, you are right then, this should be set to `nil`